jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/list/AbstractMultiGridView.java

```diff
@@ -377,6 +390,85 @@ public T getValue(T object) {
         return actionsColMeta;
     }
 
+    public List<ColumnMeta<T>> renameProcessVariables(ListTable<T> extendedPagedTable, List<ColumnMeta<T>> columnMetas) {
+        List<GridColumnPreference> columnPreferenceList = extendedPagedTable.getGridPreferencesStore().getColumnPreferences();
+
+        return (List<ColumnMeta<T>>) columnPreferenceList.stream().filter(colPref -> !isColumnAdded(columnMetas, colPref.getName()))
+                .map(colPref -> getColumnName(colPref, columnMetas))
+                .map(this::getColumnMeta)
+                .collect(Collectors.toList());
+    }
+
+    private String getColumnName(GridColumnPreference colPref, List<ColumnMeta<T>> columnMetas) {
+        String columnName = colPref.getName();
+        return existsColumnWithSameName(colPref, columnMetas) ? "Var_" + columnName : columnName;
+    }
+
+    private ColumnMeta<T> getColumnMeta(String columnName) {
+        Column genericColumn = initGenericColumn(columnName);
+        genericColumn.setSortable(false);
+        return new ColumnMeta<T>(genericColumn,
+                                 columnName,
+                                 true,
+                                 true);
+    }
+
+    public void addDomainSpecifColumns(ExtendedPagedTable<T> extendedPagedTable,
+                                       Set<String> columns) {
+        extendedPagedTable.storeColumnToPreferences();
+
+        removeRedundantColumns(extendedPagedTable, columns);
+
+        List<ColumnMeta<T>> columnMetaList = renameDomainSpecifColumns(extendedPagedTable,
+                                                                       columns);
+
+        addColumns(columnMetaList,columns);
+
+        extendedPagedTable.addColumns(columnMetaList);
+    }
+
+    private void addColumns(List<ColumnMeta<T>> columnMetaList,
```

Comment:
@bxf12315 The addColumns method is not very readable, mainly because of the long spaghetti code on the first line. Also, as I mentioned previously the logic in forEach is similar that the one in map of the renameDomainSpecificColumns. This is how I refactored these two methods:

```
private void addColumns(List<ColumnMeta<T>> columnMetaList,
                            Set<String> newColumns) {
        newColumns.stream()
                .filter(newColumn -> newColumnIsNotInDataStoreNames(newColumn, columnMetaList))
                .map(this::createNewColumn)
                .forEach(columnMetaList::add);
    }

    private List<ColumnMeta<T>> renameDomainSpecifColumns(ExtendedPagedTable<T> extendedPagedTable, Set<String> columns) {
        return getColumnsForRenaming(extendedPagedTable, columns).stream()
                .map(column -> updateColumn(column, "Var_" + column))
                .collect(Collectors.toList());
    }

    private boolean newColumnIsNotInDataStoreNames(String newColumn, List<ColumnMeta<T>> columnMetaList) {
        return columnMetaList.stream()
                .map(colMeta -> colMeta.getColumn().getDataStoreName())
                .noneMatch(newColumn::contains);
    }

    private List<String> getColumnsForRenaming(ExtendedPagedTable<T> extendedPagedTable, Set<String> columns) {
        return extendedPagedTable.getColumnMetaList().stream()
                .filter(colMeta -> !colMeta.isExtraColumn())
                .map(ColumnMeta::getCaption)
                .filter(columns::contains)
                .collect(Collectors.toList());
    }

    private ColumnMeta<T> createNewColumn(String newName) {
        return updateColumn(newName, newName);
    }

    private ColumnMeta<T> updateColumn(String columnName, String newName) {
        Column genericColumn = initGenericColumn(columnName);
        genericColumn.setSortable(false);
        return new ColumnMeta<T>(genericColumn,
                                 newName,
                                 true,
                                 true));
    }
```
Of course, you have a better insight on what it actually does, so you can come up with better names than `newColumnIsNotInDataStoreNames` or `getColumnsForRenaming`. The idea is not to have long hard-to-read streams but instead split it in more methods that descibe what you are actually doing. And these method are also useful to not have to write similar code again.

Reply:
updated,